postreise/plot/plot_bar_generation_vs_capacity.py

```diff
@@ -0,0 +1,303 @@
+):
+    """Plot any number of scenarios as bar or horizontal bar charts with two columns per
+    scenario - generation and capacity.
+
+    :param list/str areas: list of area(s), each area is one of *loadzone*, *state*,
+        *state abbreviation*, *interconnect*, *'all'*
+    :param list/str area_types: list of area_type(s), each area_type is one of
+        *'loadzone'*, *'state'*, *'state_abbr'*, *'interconnect'*, defaults to None.
+    :param int/list/str scenario_ids: list of scenario id(s), defaults to None.
+    :param list/str scenario_names: list of scenario name(s) of same len as scenario
+        ids, defaults to None
+    :param tuple time_range: [start_timestamp, end_timestamp] where each time stamp
+        is pandas.Timestamp/numpy.datetime64/datetime.datetime. If None, the entire
+        time range is used for the given scenario.
+    :param str time_zone: new time zone, defaults to None, which uses UTC.
+    :param list custom_data: list of dictionaries with each element being
+        hand-generated data as returned by
+        :func:`postreise.plot_bar_generation_vs_capacity.make_gen_cap_custom_data`,
+        defaults to None.
+    :param list/str resource_types: list of resource type(s) to show, defaults to None,
+        which shows all available resources in the area of the corresponding scenario.
+    :param dict resource_labels: a dictionary with keys being resource_types and values
+        being labels to show in the plots, defaults to None, which uses
+        resource_types as labels.
+    :param bool horizontal: display bars horizontally, default to False
+    """
+    if isinstance(areas, str):
+        areas = [areas]
+    if isinstance(area_types, str):
+        area_types = [area_types]
+    if not area_types:
+        area_types = [None] * len(areas)
+    if len(areas) != len(area_types):
+        raise ValueError(
+            "ERROR: if area_types are provided, it should have the same number of entries with areas."
+        )
+
+    if not scenario_ids:
+        scenario_ids = []
+    if isinstance(scenario_ids, (int, str)):
+        scenario_ids = [scenario_ids]
+    if isinstance(scenario_names, str):
+        scenario_names = [scenario_names]
+    if scenario_names and len(scenario_names) != len(scenario_ids):
+        raise ValueError(
+            "ERROR: if scenario names are provided, number of scenario names must match number of scenario ids"
+        )
+    if not custom_data:
+        custom_data = {}
+    if len(scenario_ids) + len(custom_data) <= 1:
+        raise ValueError(
+            "ERROR: must include at least two scenario ids and/or custom data"
+        )
+    if isinstance(resource_types, str):
+        resource_types = [resource_types]
+    if not resource_labels:
+        resource_labels = dict()
+    if not isinstance(resource_labels, dict):
+        raise TypeError("ERROR: resource_labels should be a dictionary")
+
+    grid = Grid(["USA"])
+    id2loadzone = grid.id2zone
```

Comment:
We will have to refactor that once we have the grid model selection feature merged in